### PR TITLE
Fix Table aria attributes

### PR DESCRIPTION
**Pull request by lazytype**
## ✍️ Proposed changes

https://jira.mongodb.org/browse/PD-1072
https://jira.mongodb.org/browse/PD-1073

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

File: packages/table/src/Row.tsx [hunk 1]
```diff
@@ -190,17 +210,24 @@ const Row = React.forwardRef(
...
+        const Icon = isExpanded ? ChevronDownIcon : ChevronRightIcon;
+
+        const chevronButton = (
+          <IconButton
+            onClick={() => setIsExpanded(curr => !curr)}
+            aria-label={isExpanded ? 'Collapse row' : 'Expand row'}
+            aria-expanded={isExpanded}
+            className={iconButtonMargin}
+          >
+            <Icon aria-hidden color={uiColors.gray.dark2} />
```
Comment (brookescarlett) [inline packages/table/src/Row.tsx:228]: 👍

File: packages/table/src/TableHeader.tsx [hunk 2]
```diff
@@ -107,14 +112,8 @@ function TableHeader<Shape>({
...
-  let glyph: 'unsorted' | 'asc' | 'desc';
-
-  if (sort?.columnId === index) {
-    glyph = sort?.direction ?? 'unsorted';
-  } else {
-    glyph = 'unsorted';
-  }
-
+  const sortDirection = sort && sort.columnId === index ? sort.direction : null;
+  const glyph: 'unsorted' | 'asc' | 'desc' = sortDirection ?? 'unsorted';
```
Comment (brookescarlett) [inline packages/table/src/TableHeader.tsx:116]: should we create an enum for these values?

File: packages/table/src/Row.tsx [hunk 3]
```diff
@@ -190,17 +210,24 @@ const Row = React.forwardRef(
             React.cloneElement(child, {
               children: <span>{child.props.children}</span>,
               key: `${indexRef.current}-${index}`,
-              disabled: child.props.disabled || disabled,
```
Comment (brookescarlett) [inline packages/table/src/Row.tsx:193]: Why rm?

Comment (lazytype) [reply packages/table/src/Row.tsx:193]: The `td`/`th` elements don't have a `disabled` attribute. Typing the `isComponentType` call revealed this

Comment (brookescarlett) [reply packages/table/src/Row.tsx:193]: Ah, good to know 😅

File: packages/table/src/TableHeader.tsx (hunk 2 above)
Comment (lazytype) [reply packages/table/src/TableHeader.tsx:116]: something we can consider doing if we use these values in more places

**Review by brookescarlett — APPROVED** [review]

**Pull request merged**